stripe/src/main/java/com/stripe/android/SetupIntentResult.java

```diff
@@ -0,0 +1,69 @@
+package com.stripe.android;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.stripe.android.model.SetupIntent;
+import com.stripe.android.model.StripeIntent;
+import com.stripe.android.utils.ObjectUtils;
+
+public class SetupIntentResult implements StripeIntentResult<SetupIntent> {
+    @NonNull private final SetupIntent mSetupIntent;
+    @Status private final int mStatus;
+
+    private SetupIntentResult(@NonNull Builder builder) {
+        mSetupIntent = builder.mSetupIntent;
+        mStatus = builder.mStatus;
+    }
+
+    @NonNull
+    @Override
+    public SetupIntent getIntent() {
+        return mSetupIntent;
+    }
+
+    @Override
+    public int getStatus() {
+        return mStatus;
+    }
+
+    static final class Builder<T extends StripeIntent>
```

Comment:
feels weird that we never use `T` or `StripeIntent`, I don't have a better suggestion, just a bit of code smell